wsd/RequestDetails.cpp

```diff
@@ -283,17 +284,34 @@ Poco::URI RequestDetails::sanitizeURI(const std::string& uri)
 
 std::string RequestDetails::getDocKey(const Poco::URI& uri)
 {
-    // If multiple host-names are used to access us, then
-    // they must be aliases. Permission to access aliased hosts
-    // is checked at the point of accepting incoming connections.
-    // At this point storing the hostname artificially discriminates
-    // between aliases and forces same document (when opened from
-    // alias hosts) to load as separate documents and sharing doesn't
-    // work. Worse, saving overwrites one another.
     std::string docKey;
-    Poco::URI::encode(uri.getPath(), "", docKey);
-    LOG_INF("DocKey from URI [" << uri.toString() << "] => [" << docKey << ']');
+    std::string uriHost = uri.getHost();
+    std::string uriPort = std::to_string(uri.getPort());
+
+    RequestDetails::setUriHostAndPort(uriHost, uriPort);
+
+    Poco::URI::encode(uri.getScheme() + "://" + uriHost + ":" + uriPort + uri.getPath(), "",docKey);
+    LOG_INF("DocKey from URI [" << uri.toString() << "] => [" << docKey << "]");
     return docKey;
 }
 
+void RequestDetails::setUriHostAndPort(std::string& uriHost, std::string& uriPort)
+{
+    if (StorageBase::realHosts.find(uriHost) == StorageBase::realHosts.end())
+    {
+        const std::string key = uriHost + ":" + uriPort;
+        if (StorageBase::aliasHosts.find(uriHost) != StorageBase::aliasHosts.end() &&
+            StorageBase::aliasPorts.find(key) != StorageBase::aliasPorts.end())
+        {
+            uriPort = StorageBase::aliasPorts[key];
+            uriHost = StorageBase::aliasHosts[uriHost];
+        }
+        else
+        {
+            LOG_ERR("hostname:" << uriHost
+                                << " does not exists in aliases configuration in coolwsd.xml, "
```

Comment:
Generally not a good idea to refer to filenames in the log. This simply makes it harder to change the filename later. Simply referring to "configuration" should be enough.

And it is not clear why this is an error. The description of "fallback" says "fallback setting when alias is missing from the list," so it shouldn't be an error to have "alias" missing. This _shouldn't_ even be info-level, since this function is called frequently. I'd make it trace.

Also, a space after `hostname:` should improve readability.